internal/locald/rootmanager/tp_monitor.go

```diff
@@ -128,16 +129,40 @@ func (mon *tpMonitor) checkTunnelProxyAccess(ctx context.Context) bool {
 			restartSvcs = true
 		}
 	}
+	if !restartSvcs {
+		if mon.root == nil || mon.root.root == nil || mon.root.root.etcHostsSVC == nil || !mon.root.root.etcHostsSVC.Status().Healthy {
+			return false
+		}
+	}
+	if !restartSvcs {
```

Comment:
I think this block of code could be part of the same `if` statement from line 132.

Reply:
sure